aten/src/ATen/BatchedFallback.cpp

```diff
@@ -105,8 +105,14 @@ void batchedTensorForLoopFallback(const c10::OperatorHandle& op, torch::jit::Sta
   // Strategy: For each batch, we are going to push slices (where applicable)
   // of the arguments onto `stack`, call `op`, and store the result in
   // `output_shards`.
-  std::vector<Tensor> output_shards;
-  output_shards.reserve(num_batches);
+  //
+  // NOTE: [Output shards layout]
+  // Assume that the operator has three outputs: a, b, c.
+  // The layout of output_shards is as follows:
+  // [ a0, a1, a2, a3, b0, b1, b2, b3, c0, c1, c2, c3]
+  // This is so that we can call at::stack([a0...a3]), at::stack([b0...b3])
+  // more easily in the next step.
+  std::vector<Tensor> output_shards(num_batches * num_returns);
```

Comment:
MatrixRef may be of interest here!

Reply:
I changed the indexing code to use MatrixRef! It's a nice way to abstract that behavior, thanks for pointing it out